src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -2186,7 +2194,8 @@ private unsafe ParsingError PrivateParseMinimal()
 
                     // This will disallow '\' as the host terminator for any scheme that is not implicitFile or cannot have a Dos Path
                     if ((idx < (ushort)length && pUriString[idx] == '\\') && NotAny(Flags.ImplicitFile) &&
-                        _syntax.NotAny(UriSyntaxFlags.AllowDOSPath))
+                        (_syntax.NotAny(UriSyntaxFlags.AllowDOSPath) ||
+                        (InFact(Flags.UncPath) && _syntax.NotAny(UriSyntaxFlags.ConvertPathSlashes))))
```

Comment:
@tmds This is the behavior we want, so that file://authority/path?query#fragment is valid and file://authority\path?query#fragment will be invalid.